ethapi/api.go

```diff
@@ -998,7 +998,7 @@ func DoEstimateGas(ctx context.Context, b Backend, args TransactionArgs, blockNr
 	// Recap the highest gas limit with account's available balance.
 	if feeCap.BitLen() != 0 {
 		state, _, err := b.StateAndHeaderByNumberOrHash(ctx, blockNrOrHash)
-		if err != nil {
+		if state == nil || err != nil {
```

Comment:
StateAndHeaderByNumberOrHash returns an error if not found, but it's okay to double check